packages/react-dom/src/__tests__/utils/ReactDOMServerIntegrationTestUtils.js

```diff
@@ -51,11 +52,24 @@ module.exports = function (initModules) {
   async function asyncReactDOMRender(reactElement, domElement, forceHydrate) {
     if (forceHydrate) {
       await act(() => {
-        ReactDOM.hydrate(reactElement, domElement);
+        if (ReactDOMClient) {
+          ReactDOMClient.hydrateRoot(domElement, reactElement, {
+            onRecoverableError: () => {
+              // TODO: assert on recoverable error count.
```

Comment:
Updated to add and ignore `onRecoverableError` errors. I started to add support for asserting this count, but it's going to require a lot more thought and tests to be updated so I'm not adding it here.